docs/REST API/rest-api.md

```diff
@@ -3536,6 +3592,128 @@ Retrieves a list of the configuration profiles assigned to a host.
 }
 ```
 
+### Get host's past activity
+
+`GET /api/v1/fleet/hosts/:id/activites/past`
+
+#### Parameters
+
+| Name | Type    | In   | Description                  |
+| ---- | ------- | ---- | ---------------------------- |
+| id   | integer | path | **Required**. The host's id. |
+| page | integer | query | Page number of the results to fetch.|
+| per_page | integer | query | Results per page.|
+
+#### Example
+
+`GET /api/v1/fleet/hosts/12/activites/past`
+
+##### Default response
+
+`Status: 200`
+
+```json
+{
+  "activities": [
+    {
+      "created_at": "2023-07-27T14:35:08Z",
+      "id": 2,
+      "actor_full_name": "Anna",
+      "actor_id": 1,
+      "actor_gravatar": "",
+      "actor_email": "anna@example.com",
+      "type": "ran_script",
+      "details": {
+        "host_id": 1,
+        "host_display_name": "Steve's MacBook Pro",
+        "script_name": "set-timezones.sh",
+        "script_execution_id": "d6cffa75-b5b5-41ef-9230-15073c8a88cf",
+      },
+    },
+    {
+      "created_at": "2021-07-27T13:25:21Z",
+      "id": 1,
+      "actor_full_name": "Bob",
+      "actor_id": 2,
+      "actor_gravatar": "",
+      "actor_email": "bob@example.com",
+      "type": "ran_script",
+      "details": {
+        "host_id": 1,
+        "host_display_name": "Steve's MacBook Pro",
+        "script_name": "",
+        "script_execution_id": "y3cffa75-b5b5-41ef-9230-15073c8a88cf",
+      },
+    },
+  ],
+  "meta": {
+    "has_next_results": false,
+    "has_previous_results": false
+  }
+}
+```
+
+### Get host's upcoming activity
+
+`GET /api/v1/fleet/hosts/:id/activites/upcoming`
+
+#### Parameters
+
+| Name | Type    | In   | Description                  |
+| ---- | ------- | ---- | ---------------------------- |
+| id   | integer | path | **Required**. The host's id. |
+| page | integer | query | Page number of the results to fetch.|
+| per_page | integer | query | Results per page.|
+
+#### Example
+
+`GET /api/v1/fleet/hosts/12/activites/upcoming`
+
+##### Default response
+
+`Status: 200`
+
+```json
+{
+  "activities": [
+    {
+      "created_at": "2023-07-27T14:35:08Z",
+      "id": 3,
```

Comment:
@noahtalerman to future-proof with where we're headed - e.g. we will eventually want to support canceling pending jobs, and to make it clearer that this is not the same thing as "past" activities, i.e. an `id` here does not correspond to the same `id` there - I'd return an `uuid` here instead. For now, it would be the host script execution request uuid, and when we add more types of upcoming activities, it will fit in better (e.g. an MDM command uuid, etc.).

I'll go ahead with that assumption (and will use `uuid` as field name instead). Let me know if that works for you.

Reply:
Just hopped on a call live and decided to go w/ `uuid`.

The unified queue, when we add MDM commands, will be pulling scripts/commands from two separate tables in the DB. Need to have unique identifiers across scripts/commands.

This is nice for the user because they can see that the `uuid` corresponds to the `execution_id` for scripts. So they can use the same identifier to get results and, in the future, cancel or bump priority for the script.